### PR TITLE
rename osr to assume/else

### PR DESCRIPTION
**Pull request by o-**
based on fix-inline #167

### CODE REVIEW

File: instr.ml [hunk 1]
```diff
@@ -109,6 +109,29 @@ and binop =
...
+      | Lt   -> Gte
+      | Lte  -> Gt
+      | Gt   -> Lte
+      | Gte  -> Lt
+      | _    -> assert(false)
+      end in
+      Some (Binop (nop, e1, e2))
+  | Simple (Constant (Nil| Int _|Fun_ref _|Array _))
+  | Binop ((Plus|Sub|Mult|Div|Mod), _, _)
+  | Binop ((And|Or), _, _)  (* TODO *)
```
Comment (gasche) [inline instr.ml:130]: we could just add `NAnd` and `NOr` as binops.

**Comment by gasche** [comment]
I think that the two relevant commits 92fb88fb680d3c34568575f82114b7332ec24064 and 92fb88fb680d3c34568575f82114b7332ec24064 should be squashed together because the intermediary state is just too confusing.

**Review by gasche — COMMENTED** [review]
If I understand correctly, #167 may still need more work, but the new commits in there as fine (although they should be squashed, see comment).

File: instr.ml (hunk 1 above)
Comment (o-) [reply instr.ml:130]: yeah, but probably not priority

**Comment by mhyee** [comment]
I ran `git grep osr | grep -v doc` (to ignore all the instances in the `doc` directory)

```
check.ml:130:    (* Check correctness of calls and osrs *)
eval.ml:267:  let build_osr_frame varmap old_env old_heap =
eval.ml:274:  let build_extra_osr_frame (heap, frames) {cont_pos; cont_res; varmap} =
eval.ml:275:    let osr_env, heap = build_osr_frame varmap conf.env heap in
eval.ml:278:    (* osr target is a label, but continuation uses pc,
eval.ml:284:    heap, (cont_res, osr_env, pos) :: frames
eval.ml:398:      let osr_env, heap = build_osr_frame varmap conf.env conf.heap in
eval.ml:400:        List.fold_left build_extra_osr_frame (heap, []) (List.rev extra_frames) in
eval.ml:406:        env = osr_env;
instr.ml:468:  method osr_label l = l
instr.ml:505:        label = m#osr_label label;
instr.ml:507:        target = m#osr_target target;
instr.ml:508:        varmap = m#osr_varmap varmap;
instr.ml:520:  method osr_func_label l = l
instr.ml:521:  method osr_version_label l = l
instr.ml:522:  method osr_target_label l = m#osr_label l
instr.ml:524:  method osr_target {func; version; pos} = {
instr.ml:525:    func = m#osr_func_label func;
instr.ml:526:    version = m#osr_version_label version;
instr.ml:527:    pos = m#osr_target_label pos }
instr.ml:531:    varmap = m#osr_varmap varmap;
instr.ml:532:    cont_pos = m#osr_target cont_pos;
instr.ml:535:  method osr_varmap = List.map m#osr_varmap_entry
instr.ml:536:  method osr_varmap_entry = function
transform_assumption.ml:195: * Like in insert_assumption the strategy is to find the earliest possible osr
transform_assumption.ml:200: *   osr [x==1] ...
transform_assumption.ml:202: *   osr [x==2] ...
transform_assumption.ml:204: *   osr [x==1,x==2] ...
transform_assumption.ml:206: *   osr []
transform_assumption.ml:211: *    osr [] ...
transform_assumption.ml:215: *    osr [x==1] ...
transform_assumption.ml:220: *    osr [x==1] ...
transform_assumption.ml:224: *    osr [] ...
transform_inline.ml:186:              (* To be able to osr out of this call we need to have a var_map
```

**Pull request merged**